src/analysis/calculus/formal_multilinear_series.lean

```diff
@@ -164,3 +164,132 @@ lemma comp_formal_multilinear_series_apply'
 rfl
 
 end continuous_linear_map
+
+namespace formal_multilinear_series
+
+section order
+
+variables [comm_ring 𝕜] {n : ℕ}
+  [add_comm_group E] [module 𝕜 E] [topological_space E] [topological_add_group E]
+  [has_continuous_const_smul 𝕜 E]
+  [add_comm_group F] [module 𝕜 F] [topological_space F] [topological_add_group F]
+  [has_continuous_const_smul 𝕜 F]
+  {p : formal_multilinear_series 𝕜 E F}
+
+lemma eq_zero_iff : p = 0 ↔ ∀ n, p n = 0 :=
+by simp only [function.funext_iff, pi.zero_apply]
+
+lemma exists_ne_zero_of_ne_zero (hp : p ≠ 0) : ∃ n, p n ≠ 0 :=
+by simpa using eq_zero_iff.not.mp hp
+
+/-- The index of the first non-zero coefficient in `p` (or `0` if all coefficients are zero). This
+  is the order of the isolated zero of an analytic function `f` at a point if `p` is the Taylor
+  series of `f` at that point. -/
+noncomputable def order (p : formal_multilinear_series 𝕜 E F) : ℕ :=
+Inf { n | p n ≠ 0 }
+
+@[simp] lemma order_zero : (0 : formal_multilinear_series 𝕜 E F).order = 0 := by simp [order]
+
+lemma ne_zero_of_order_ne_zero (hp : p.order ≠ 0) : p ≠ 0 :=
+λ h, by simpa [h] using hp
+
+lemma order_eq_find [decidable_pred (λ n, p n ≠ 0)] (hp : ∃ n, p n ≠ 0) :
+  p.order = nat.find hp :=
+by simp [order, Inf, hp]
+
+lemma order_eq_find' [decidable_pred (λ n, p n ≠ 0)] (hp : p ≠ 0) :
+  p.order = nat.find (exists_ne_zero_of_ne_zero hp) :=
+order_eq_find _
+
+lemma order_eq_zero_iff (hp : p ≠ 0) : p.order = 0 ↔ p 0 ≠ 0 :=
+begin
+  classical,
+  have : ∃ n, p n ≠ 0 := exists_ne_zero_of_ne_zero hp,
+  simp [order_eq_find this, hp]
+end
+
+lemma order_eq_zero_iff' : p.order = 0 ↔ p = 0 ∨ p 0 ≠ 0 :=
+by { by_cases h : p = 0; simp [h, order_eq_zero_iff] }
+
+lemma apply_order_ne_zero (hp : p ≠ 0) : p p.order ≠ 0 :=
+begin
+  classical,
+  let h := exists_ne_zero_of_ne_zero hp,
+  exact (order_eq_find h).symm ▸ nat.find_spec h
+end
+
+lemma apply_order_ne_zero' (hp : p.order ≠ 0) : p p.order ≠ 0 :=
+apply_order_ne_zero (ne_zero_of_order_ne_zero hp)
+
+lemma apply_eq_zero_of_lt_order (hp : n < p.order) : p n = 0 :=
+begin
+  by_cases p = 0,
+  { simp [h] },
+  { classical,
+    rw [order_eq_find' h] at hp,
+    simpa using nat.find_min _ hp }
+end
+
+end order
+
+section coef
+
+variables [nontrivially_normed_field 𝕜]
+  [normed_add_comm_group E] [normed_space 𝕜 E] {s : E}
+  {p : formal_multilinear_series 𝕜 𝕜 E} {f : 𝕜 → E}
+  {n : ℕ} {z z₀ : 𝕜} {y : fin n → 𝕜}
+
+open_locale big_operators
+
+/-- The `n`th coefficient of `p` when seen as a power series. -/
+def coef (p : formal_multilinear_series 𝕜 𝕜 E) (n : ℕ) : E := p n 1
+
+@[simp] lemma apply_eq_prod_smul_coef : p n y = (∏ i, y i) • p.coef n :=
+begin
+  convert (p n).to_multilinear_map.map_smul_univ y 1,
+  funext; simp only [pi.one_apply, algebra.id.smul_eq_mul, mul_one],
+end
+
+lemma coef_eq_zero : p.coef n = 0 ↔ p n = 0 :=
+begin
+  split; intro h,
+  { ext; simp [h] },
+  { simp [coef, h] }
+end
```

Comment:
Okay this made me find a lot of API holes :sweat_smile: 

Could you add : 
- in `linear_algebra/multilinear/basic` after `mk_pi_ring_apply_one_eq_self`:
```lean
lemma mk_pi_ring_eq_zero_iff [fintype ι] (z : M₂) :
  multilinear_map.mk_pi_ring R ι z = 0 ↔ z = 0 :=
begin
  simp_rw [multilinear_map.ext_iff, mk_pi_ring_apply],
  split; intro h,
  { simpa using h (λ _, 1) },
  { intro x, simp [h] }
end
```
- in `topology/algebra/module/multilinear` after `ext`:
```lean
theorem ext_iff {f f' : continuous_multilinear_map R M₁ M₂} : f = f' ↔ ∀ x, f x = f' x :=
by rw [← to_multilinear_map_inj.eq_iff, multilinear_map.ext_iff]; refl
```
- in `analysis/normed_space/multilinear` after `mk_pi_field_apply_one_eq_self`:
```lean
lemma mk_pi_field_eq_zero_iff [fintype ι] (z : G) :
  continuous_multilinear_map.mk_pi_field 𝕜 ι z = 0 ↔ z = 0 :=
begin
  rw ← to_multilinear_map_inj.eq_iff,
  exact multilinear_map.mk_pi_ring_eq_zero_iff z
end
```
- just after the definition of `coef`:
```lean
lemma mk_pi_field_coeff_eq : continuous_multilinear_map.mk_pi_field 𝕜 (fin n) (p.coef n) = p n :=
(p n).mk_pi_field_apply_one_eq_self
```

Then you can do 
```suggestion
lemma coef_eq_zero : p.coef n = 0 ↔ p n = 0 :=
by rw [← mk_pi_field_coeff_eq p, continuous_multilinear_map.mk_pi_field_eq_zero_iff]
```

That's definitely not shorter overall, but these intermediate lemmas should really be there imo